File: BOJ(Java 공부용)/2851.java
import java.io.*;
import java.util.ArrayList;
import java.util.Collections;

import static java.lang.Integer.max;
import static java.lang.StrictMath.abs;

public class Main {
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    static int[] gox = {0, 1, -1, 0, 1, 1, -1, -1};
    static int[] goy = {1, 0, 0, -1, 1, -1, 1, -1};

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        ArrayList<Integer> v = new ArrayList<>();
        int sum = 0;
        int[] arr = new int[10];
        for (int i = 0; i < 10; i++) {
            arr[i] = Integer.parseInt(br.readLine());
            sum += arr[i];
            v.add(sum);
        }

        Collections.sort(v);
        int min = (int) 1e9, num = 0;

        for (int i = 0; i < v.size(); i++) {
            if (min > abs(100 - v.get(i))) {
                min = abs(100 - v.get(i));
                num = v.get(i);
            } else if (min == abs(v.get(i) - 100)) num = max(num, v.get(i));
        }
        System.out.println(num);
    }

}

File: BOJ(Python 공부용)/2851.py
import sys

sys.setrecursionlimit(50000000)

arr = [0] * 10
v = [0]*10
sum = 0
for i in range(10):
    arr[i] = int(sys.stdin.readline())
    sum += arr[i]
    v.append(sum)

v.sort()
min = int(1e9)
num = 0

for i in range(len(v)):
    if min > abs(100 - v[i]):
        min = abs(100 - v[i])
        num = v[i]
    elif min == abs(v[i] - 100):
        num = max(num, v[i])

print(num)
